src/gamma_inc.jl

```diff
@@ -0,0 +1,599 @@
+
+#----------------COEFFICIENTS FOR TEMME EXPANSION------------------
+
+d00 = -.333333333333333E+00
+d0=[.833333333333333E-01 , -.148148148148148E-01 , .115740740740741E-02 , .352733686067019E-03 , -.178755144032922E-03 , .391926317852244E-04]
+d10 = -.185185185185185E-02
+d1=[-.347222222222222E-02 , .264550264550265E-02 , -.990226337448560E-03 , .205761316872428E-03]
+d20 = .413359788359788E-02
+d2=[-.268132716049383E-02 , .771604938271605E-03]
+d30 = .649434156378601E-03
+d3=[.229472093621399E-03 , -.469189494395256E-03]
+d40 = -.861888290916712E-03
+d4=[.784039221720067E-03]
+d50 = -.336798553366358E-03
+d5=[-.697281375836586E-04]
+d60 = .531307936463992E-03
+d6=[-.592166437353694E-03]
+d70 = .344367606892378E-03
+d80 = -.652623918595309E-03
+
+# The kernel of log1pmx
+# Accuracy within ~2ulps for -0.227 < x < 0.315
+function _log1pmx_ker(x::Float64)
+    r = x/(x+2.0)
+    t = r*r
+    w = @horner(t,
+                6.66666666666666667e-1, # 2/3
+                4.00000000000000000e-1, # 2/5
+                2.85714285714285714e-1, # 2/7
+                2.22222222222222222e-1, # 2/9
+                1.81818181818181818e-1, # 2/11
+                1.53846153846153846e-1, # 2/13
+                1.33333333333333333e-1, # 2/15
+                1.17647058823529412e-1) # 2/17
+    hxsq = 0.5*x*x
+    r*(hxsq+w*t)-hxsq
+end
+"""
+    logmxp1(x::Float64)
+Return `log(x) - x + 1` carefully evaluated.
+"""
+function logmxp1(x::Float64)
+    if x <= 0.3
+        return (log(x) + 1.0) - x
+    elseif x <= 0.4
+        u = (x-0.375)/0.375
+        return _log1pmx_ker(u) - 3.55829253011726237e-1 + 0.625*u
+    elseif x <= 0.6
+        u = 2.0*(x-0.5)
+        return _log1pmx_ker(u) - 1.93147180559945309e-1 + 0.5*u
+    else
+        return log1pmx(x - 1.0)
+    end
+end
+
+"""
+   rgamma1pm1(a)
+
+   Computation of 1/Gamma(a+1) - 1 for -0.5<=a<=1.5 : ``1/\\Gamma (a+1) - 1``
+"""
+function rgamma1pm1(a::Float64)
+    t=a
+    d = a - 0.5
+    if d > 0.0
+        t = d - 0.5
+    end
+    if t == 0.0
+        return 0.0
+    elseif t < 0.0
+     top = @horner(t , -.422784335098468E+00 , -.771330383816272E+00 , -.244757765222226E+00 , .118378989872749E+00 , .930357293360349E-03 , -.118290993445146E-01 , .223047661158249E-02 , .266505979058923E-03 , -.132674909766242E-03)
+     bot = @horner(t , 1.0 , .273076135303957E+00 , .559398236957378E-01)
+     w = top/bot
+     if d > 0.0
+        return t*w/a
+     else
+        return a*(w + 1.0)
+     end
+    else
+     top = @horner(t , .577215664901533E+00 , -.409078193005776E+00 , -.230975380857675E+00 , .597275330452234E-01 , .766968181649490E-02 , -.514889771323592E-02 , .589597428611429E-03)
+     bot = @horner(t , 1.0 , .427569613095214E+00 , .158451672430138E+00 , .261132021441447E-01 , .423244297896961E-02) 
+     w = top/bot
+     if d > 0.0
+        return (t/a)*(w - 1.0)
+     else
+        return a*w
+     end
+    end
+end
+"""
+    exp1xgamma1(a,x)
```

Comment:
I'm not sure if there is a standard name for this. Didonato just calls it `R`. Other authors use `D` for a similar function (with the denominator being `gamma(a+1)` instead of `gamma(a)`).

It is closely related to the gamma and poisson pdf functions, so maybe something like `rgammax` for now? (there is a bit of a precedence for suffixing with `x` when scaled by a factor, see `besseljx`, `erfcx` etc)

Reply:
Okay fine!